src/app/Services/User/Import.php

```diff
@@ -0,0 +1,277 @@
+<?php
+/*
+ * This code was developed by NIMA SOFTWARE SRL | nimasoftware.com
+ * For details contact contact@nimasoftware.com
+ */
+
+
+namespace App\Services\User;
+
+
+use App\County;
+use App\Institution;
+use App\Organisation;
+use App\User;
+use Doctrine\CouchDB\CouchDBException;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Support\Str;
+
+class Import
+{
+
+    private $rolesList = [];
+
+    private $countiesMap = [];
+
+    private $errors = [];
+
+    private $citiesMap = [];
+
+    private $usersCreated = [];
+
+    public function __construct()
+    {
+        $this->setup();
+    }
+
+    public function setup()
+    {
+        $this->rolesList = array_values(config('roles.role'));
+        $this->prepareCountiesMap();
+    }
+
+    protected function prepareCountiesMap()
+    {
+        /** @var \Illuminate\Database\Eloquent\Collection $countys */
+        $counties = County::all(['_id', "slug", "name"]);
+        $this->countiesMap = array_column($counties->toArray(), null, 'slug');
+    }
+
+    public function importRescueOfficers(iterable $list, User $importedBy)
+    {
+        return $this->import($list, User::ROLE_RESCUE_OFFICER, $importedBy);
+    }
+
+    public function import(iterable $dataset, $userRole, User $importedBy): self
+    {
+        if (!in_array($userRole, $this->rolesList)) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'Invalid user type! Must be one of  %s you provided %s',
+                    print_r($this->rolesList, true),
+                    $userRole
+                )
+            );
+        }
+
+
+        /*
+         * array(7) {
+                  [0]=> "test popescu"
+                  [1]=> "asfa@fas.asd"
+                  [2]=> "Iasi"
+                  [3]=> "Tg. frumos"
+                  [4]=> "123123"
+                  [5]=> "institutie slug"
+                  [6]=> "organizatie slug"
+                }
```

Comment:
To make the code more easy to read you can replace this comment with some consts and use the consts in the following code:
const NAME = 0;
const EMAIL = 1;
const COUNTY = 2;
const CITY = 3;
const PHONE = 4;
const INSTITUTION = 5;
const ORGANIZATION = 6;